applications/tari_console_wallet/src/automation/commands.rs

```diff
@@ -284,60 +283,148 @@ pub async fn make_it_rain(
+    let negotiated = match args[6].clone() {
+        Negotiated(val) => Ok(val),
+        _ => Err(CommandError::Argument),
+    }?;
+
+    let message = match args[7].clone() {
         Text(m) => Ok(m),
         _ => Err(CommandError::Argument),
     }?;
 
-    // Wait until specified test start time
-    let now = Utc::now();
-    let delay_ms = if start_time > now {
-        println!(
-            "`make-it-rain` scheduled to start at {}: msg \"{}\"",
-            start_time, message
-        );
-        (start_time - now).num_milliseconds() as u64
-    } else {
-        0
-    };
+    // We are spawning this command in parallel, thus not collecting transaction IDs
+    tokio::task::spawn(async move {
+        // Wait until specified test start time
+        let now = Utc::now();
+        let delay_ms = if start_time > now {
+            println!(
+                "`make-it-rain` scheduled to start at {}: msg \"{}\"",
+                start_time, message
+            );
+            (start_time - now).num_milliseconds() as u64
+        } else {
+            0
+        };
 
-    debug!(
-        target: LOG_TARGET,
-        "make-it-rain delaying for {:?} ms - scheduled to start at {}", delay_ms, start_time
-    );
-    delay_for(Duration::from_millis(delay_ms)).await;
+        debug!(
+            target: LOG_TARGET,
+            "make-it-rain delaying for {:?} ms - scheduled to start at {}", delay_ms, start_time
+        );
+        delay_for(Duration::from_millis(delay_ms)).await;
 
-    let num_txs = (txps * duration as f64) as usize;
+        let num_txs = (txps * duration as f64) as usize;
+        let started_at = Utc::now();
 
-    let mut tx_ids = Vec::new();
-    let started_at = Utc::now();
-
-    for i in 0..num_txs {
-        // Manage Tx rate
-        let actual_ms = (Utc::now() - started_at).num_milliseconds();
-        let target_ms = (i as f64 / (txps / 1000.0)) as i64;
-        if target_ms - actual_ms > 0 {
-            // Maximum delay between Txs set to 120 s
-            delay_for(Duration::from_millis((target_ms - actual_ms).min(120_000i64) as u64)).await;
+        struct TransactionSendStats {
+            i: usize,
+            tx_id: Result<TxId, CommandError>,
+            delayed_for: Duration,
+            submit_time: Duration,
         }
-        // Send Tx
-        let amount = start_amount + inc_amount * (i as u64);
-        let send_args = vec![
-            ParsedArgument::Amount(amount),
-            ParsedArgument::PublicKey(public_key.clone()),
-            ParsedArgument::Text(message.clone()),
-        ];
-        let tx_service = wallet_transaction_service.clone();
-        let tx_id = handle
-            .spawn(send_tari(tx_service, send_args))
-            .await
-            .map_err(CommandError::Join)??;
-
-        debug!(target: LOG_TARGET, "make-it-rain tx_id: {}", tx_id);
-        tx_ids.push(tx_id);
-    }
+        let transaction_type = if negotiated { "negotiated" } else { "one-sided" };
```

Comment:
nit: This is a really long method, it would be worth your while splitting it up and simplifying it